tests/test_distributed_load_hf.py

```diff
@@ -0,0 +1,174 @@
+import json
+import os
+import pathlib
+import shutil
+import uuid
+from typing import *
+
+import pytest
+import torch
+import torch.distributed as dist
+import transformers
+
+from realhf.api.core.config import ModelFamily
+from realhf.api.core.model_api import HF_MODEL_FAMILY_REGISTRY, ReaLModelConfig
+from realhf.base import constants, logging
+from realhf.base.testing import (
+    LocalMultiProcessTest,
+    clear_name_resolve,
+    init_global_constants,
+)
+
+logger = logging.getLogger("tests.test_saveload")
+
+
+def _load_all_pytorch_bin(path: pathlib.Path):
+    if os.path.exists(path / "pytorch_model.bin.index.json"):
+        with open(path / "pytorch_model.bin.index.json", "r") as f:
+            hf_sd_mapping = json.load(f)["weight_map"]
+        sd = {}
+        for fn in hf_sd_mapping.values():
+            sd.update(torch.load(path / fn, map_location="cpu"))
+    else:
+        sd = torch.load(path / "pytorch_model.bin", map_location="cpu")
+    return sd
+
+
+def _save_then_load(
+    tmp_path: pathlib.Path,
```

Comment:
`pytest` has its own temporary directories and files management, you do not need to make tmp_path of your own. See [this](https://docs.pytest.org/en/stable/how-to/tmp_path.html).

Reply:
fixed